Tools/CMake/AMReX_SetupCUDA.cmake

```
@@ -86,6 +151,20 @@ foreach (_item IN LISTS _nvcc_arch_flags)
 
 endforeach ()
 
+if (CUDA_LTO)
+    # we replace
+    #   -gencode=arch=compute_NN,code=sm_NN
+    # with
+    #   -gencode=arch=compute_NN,code=lto_NN
+    set(_nvcc_arch_flags_org ${_nvcc_arch_flags})
+    foreach (_item IN LISTS _nvcc_arch_flags_org)
+       string(REGEX MATCH "[0-9]+" _cuda_compute_capability "${_item}")
+       string(REPLACE "code=sm_${_cuda_compute_capability}"
+                      "code=lto_${_cuda_compute_capability}"
```

I squashed this out, but in order to trigger the linker error you can do:
```suggestion
       string(REPLACE "${_item}"
                      "-dlto -arch=sm_${_cuda_compute_capability}"
```

---

The build as in https://warpx.readthedocs.io/en/latest/building/cmake.html#configure-your-compiler

```bash
cmake .. -DWarpX_COMPUTE=CUDA -DWarpX_amrex_repo=https://github.com/ax3l/amrex.git -DWarpX_amrex_branch=topic-moreCUDAcmakeFlags -DCUDA_LTO=ON

make VERBOSE=ON
```